client/src/sim.rs

```diff
@@ -377,7 +377,10 @@ impl Sim {
             metrics::declare_ready_for_profiling();
         }
         for node in &msg.nodes {
-            self.graph.insert_neighbor(node.parent, node.side);
+            // We need to get a list of nodes from the server, especially on first log-in,
+            // since otherwise, we won't be able to know where the local character is with
+            // just the NodeId alone.
+            self.graph.ensure_neighbor(node.parent, node.side);
```

Comment:
What's the significance of the change from `insert_neighbor`?

Reply:
Since the client can choose to generate nodes, we can no longer guarantee that the server won't send a node the client already knows about.